src/org/labkey/idri/idriController.java

```diff
@@ -1,517 +1,517 @@
+        {
+            ApiSimpleResponse resp = new ApiSimpleResponse();
+            PipeRoot root = PipelineService.get().findPipelineRoot(getContainer());
+
+            String containerPath = null;
+            String webdavURL = null;
+
+            if (null != root)
+            {
+                containerPath = root.getContainer().getPath();
+                webdavURL = root.getWebdavURL();
+                if (HPLCAssayDataHandler.NAMESPACE.length() > 0)
+                    webdavURL += HPLCAssayDataHandler.NAMESPACE;
+            }
+
+            resp.put("containerPath", containerPath);
+            resp.put("webDavURL", webdavURL);
+
+            return resp;
+        }
+    }
+
+    public static class HPLCResourceForm
+    {
+        public String _path;
+        public boolean _test = false;
+
+        public String getPath()
+        {
+            return _path;
+        }
+
+        public void setPath(String path)
+        {
+            _path = path;
+        }
+
+        public boolean isTest()
+        {
+            return _test;
+        }
+
+        public void setTest(boolean test)
+        {
+            _test = test;
+        }
+    }
+
+    @RequiresPermission(ReadPermission.class)
+    public class getHPLCResourceAction extends ReadOnlyApiAction<HPLCResourceForm>
+    {
+        @Override
+        public ApiResponse execute(HPLCResourceForm form, BindException errors) throws Exception
+        {
+            String path = form.getPath();
+            WebdavResource resource = WebdavService.get().lookup(path);
+            Map<String, String> props = new HashMap<>();
+            Container c = getContainer();
+
+            if (null != resource)
+            {
+                FileContentService svc = FileContentService.get();
+                ExpData data = svc.getDataObject(resource, c);
+
+                if (form.isTest() && data == null)
+                {
+                    // Generate the data to help the test, replicating what DavController would do for us
+                    File file = resource.getFile();
+                    if (null != file)
+                    {
+                        data = ExperimentService.get().createData(c, new DataType("UploadedFile"));
+                        data.setName(file.getName());
+                        data.setDataFileURI(file.toURI());
+
+                        int scale = ExperimentService.get().getTinfoData().getColumn("DataFileURL").getScale();
+                        String dataFileURL = data.getDataFileUrl();
+                        if (dataFileURL != null && dataFileURL.length() > scale)
+                        {
+                            // If the path is too long to store, bail out without creating an exp.data row
+                        }
+                        else
+                        {
+                            data.save(getUser());
+                        }
```

Comment:
@labkey-adam 
Is this ok as is or should it be a MutatingApiAction? 
(There is data saving for test (!?), so it could be a MutatingApiAction, but I am not too clear on the context here, plus it has ReadPermission)

Reply:
Same as the SignalData case... code was probably copied from here. Since Nick is gone, @labkey-ians is your best bet.

Reply:
As per chat with @labkey-jeckels, In order to get this changed merged to release19.1-SNAPSHOT before tomorrow, we are making this a MutatingApiAction, and updating the caller with POST (looks like Ankur has done this already).
An issue will be created for Ian to split this action for GET and POST/update the permission on the actions/update the callers.